src/doc/src/reference/specifying-dependencies.md

```diff
@@ -2,88 +2,113 @@
+[Caret requirement](#caret-requirements) rules.
 
-Here are some more examples of version requirements and the versions that would
-be allowed with them:
+In this case, if we ran `cargo update -p time`, cargo could update the crate to
+version `0.1.13` if it is the latest `0.1.z` release, but would not update us
+to `0.2.0`.
 
-```notrust
-1.2.3  :=  >=1.2.3, <2.0.0
-1.2    :=  >=1.2.0, <2.0.0
-1      :=  >=1.0.0, <2.0.0
-0.2.3  :=  >=0.2.3, <0.3.0
-0.2    :=  >=0.2.0, <0.3.0
-0.0.3  :=  >=0.0.3, <0.0.4
-0.0    :=  >=0.0.0, <0.1.0
-0      :=  >=0.0.0, <1.0.0
-```
+There are five types of version requirements:
 
-This compatibility convention is different from SemVer in the way it treats
-versions before 1.0.0. While SemVer says there is no compatibility before
-1.0.0, Cargo considers `0.x.y` to be compatible with `0.x.z`, where `y ≥ z`
-and `x > 0`.
+| Type                                   | Operators         | Examples                               |
+|----------------------------------------|-------------------|----------------------------------------|
+| [Caret](#caret-requirements)           | None or `^`       | `"1.2.3"` and `"^1.2.3"`               |
+| [Tilde](#tilde-requirements)           | `~`               | `"~1.2.3"`                             |
+| [Wildcard](#wildcard-requirements)     | `*`               | `"1.2.*"`                              |
+| [Comparison](#comparison-requirements) | `>`, `<` and `>=` | `">1.2.3"`, `"<1.2.3"` and `">=1.2.3"` |
+| [Strict](#strict-requirements)         | `=`               | `"= 1.2.3"`                            |
 
-It is possible to further tweak the logic for selecting compatible versions
-using special operators, though it shouldn't be necessary most of the time.
+[SemVer]: https://doc.rust-lang.org/cargo/reference/resolver.html#semver-compatibility
 
 ### Caret requirements
 
-**Caret requirements** are an alternative syntax for the default strategy,
-`^1.2.3` is exactly equivalent to `1.2.3`.
+**Caret requirements** allow SemVer compatible updates.
+
+Updates are only allowed if the new version number does not modify the left-most
+non-zero digit.
+
+This compatibility convention is different from [SemVer] in the way it treats
+versions before `"1.0.0"`. While SemVer says there is no compatibility before
+1.0.0, Cargo considers `"0.x.y"` to be compatible with `"0.x.z"`, where `y ≥ z`
+and `x` is not zero.
+
+#### Examples:
+
+If we specify the version string as `"1.4.8"`, compatible versions are:
+- `"1.4.x"` with `x ≥ 8`.
+- `"1.x.y"` with `x > 2`, `y` can be anything.
+
+This table shows all possible ways to specify a caret requirements, and the
+ranges that are considered compatible for each:
+
+```notrust
+^1      :=  >=1.0.0, <2.0.0
+^1.2    :=  >=1.2.0, <2.0.0
+^1.2.3  :=  >=1.2.3, <2.0.0
+^0      :=  >=0.0.0, <1.0.0
+^0.0    :=  >=0.0.0, <0.1.0
+^0.0.3  :=  >=0.0.3, <0.0.4 (exact)
+^0.2    :=  >=0.2.0, <0.3.0
+^0.2.3  :=  >=0.2.3, <0.3.0
```

Comment:
IIUC https://github.com/rust-lang/cargo/pull/10158 tried not to mention much about the `^` syntax on requirement to reduce confusion. Do we want to follow the rule here and throughout this PR?

Reply:
How could I proceed here?

1. Rename `caret requirement` to `SemVer requirement`.
2. Inside of the `caret requirements` section, remove all carets, and add a note explaining that `"1.4.8" == "^1.4.8"`.
3. Try to move all `caret requirements` out of it's own section, and put it on the start of the file.

I tried approaching **3** like #10158 but I found it to be weird with the new table with all the operators.

Reply:
Hmm... maybe keep calling it caret but only mention it as a alternative syntax at the end of the "Caret requirements" section? Such like

> Note that Cargo favors specifying caret requirements without a real caret `^`. You can still use the alternative syntax. For example, `^1.2.3` is exactly equivalent to `1.2.3`.

Though it still seems a bit awkward...

Reply:
I like @weihanglo's idea here to not use `^` in this section where possible, and at the end there can be a small code-block that says that `^1.2.3 := 1.2.3` or similar